packaging/installer/kickstart-static64.sh

```diff
@@ -224,56 +228,81 @@ NETDATA_INSTALLER_OPTIONS=""
-  elif [ "${1}" = "--local-files" ]; then
-    NETDATA_UPDATES="" # Disable autoupdates if using pre-downloaded files.
-    shift 1
-    if [ -z "${1}" ]; then
-      fatal "Option --local-files requires extra information. The desired tarball full filename is needed"
-    fi
+  case "${1}" in
+    "--dont-wait") opts="${opts} --accept" ;;
+    "--non-interactive") opts="${opts} --accept" ;;
+    "--accept") opts="${opts} --accept" ;;
+    "--dont-start-it")
+      NETDATA_INSTALLER_OPTIONS="${NETDATA_INSTALLER_OPTIONS:+${NETDATA_INSTALLER_OPTIONS} }${1}"
+      NETDATA_CLAIM_EXTRA="${NETDATA_CLAIM_EXTRA} -daemon-not-running"
+      ;;
+    "--no-updates") NETDATA_UPDATES="" ;;
+    "--stable-channel")
+      RELEASE_CHANNEL="stable"
+      NETDATA_INSTALLER_OPTIONS="${NETDATA_INSTALLER_OPTIONS:+${NETDATA_INSTALLER_OPTIONS} }${1}"
+      ;;
+    "--disable-telemetry") NETDATA_INSTALLER_OPTIONS="${NETDATA_INSTALLER_OPTIONS:+${NETDATA_INSTALLER_OPTIONS} }${1}";;
+    "--local-files")
+      NETDATA_UPDATES="" # Disable autoupdates if using pre-downloaded files.
+      if [ -z "${2}" ]; then
+        fatal "Option --local-files requires extra information. The desired tarball full filename is needed"
+      fi
 
-    NETDATA_LOCAL_TARBALL_OVERRIDE="${1}"
-    shift 1
-    if [ -z "${1}" ]; then
-      fatal "Option --local-files requires a pair of the tarball source and the checksum file"
-    fi
+      NETDATA_LOCAL_TARBALL_OVERRIDE="${2}"
 
-    NETDATA_LOCAL_TARBALL_OVERRIDE_CHECKSUM="${1}"
-    shift 1
-  elif [ "${1}" = "--allow-duplicate-install" ]; then
-    NETDATA_ALLOW_DUPLICATE_INSTALL=1
-    shift 1
-  elif [ "${1}" = "--reinstall" ]; then
-    NETDATA_REINSTALL=1
-    shift 1
-  else
-    echo >&2 "Unknown option '${1}' or invalid number of arguments. Please check the README for the available arguments of ${0} and try again"
-    exit 1
-  fi
+      if [ -z "${3}" ]; then
+        fatal "Option --local-files requires a pair of the tarball source and the checksum file"
+      fi
+
+      NETDATA_LOCAL_TARBALL_OVERRIDE_CHECKSUM="${3}"
+      shift 2
+      ;;
+    "--allow-duplicate-install") NETDATA_ALLOW_DUPLICATE_INSTALL=1 ;;
+    "--reinstall") NETDATA_REINSTALL=1 ;;
+    "--claim-token")
+      NETDATA_CLAIM_TOKEN="${2}"
+      shift 1
+      ;;
+    "--claim-rooms")
+      NETDATA_CLAIM_ROOMS="${2}"
+      shift 1
+      ;;
+    "--claim-url")
+      NETDATA_CLAIM_URL="${2}"
+      shift 1
+      ;;
+    "--claim-proxy")
+      NETDATA_CLAIM_EXTRA="${NETDATA_CLAIM_EXTRA} -proxy ${2}"
+      shift 1
+      ;;
+    *)
+      echo >&2 "Unknown option '${1}' or invalid number of arguments. Please check the README for the available arguments of ${0} and try again"
+      exit 1
+  esac
+  shift 1
 done
 
 if [ ! "${DO_NOT_TRACK:-0}" -eq 0 ] || [ -n "$DO_NOT_TRACK" ]; then
   NETDATA_INSTALLER_OPTIONS="${NETDATA_INSTALLER_OPTIONS:+${NETDATA_INSTALLER_OPTIONS} }--disable-telemtry"
 fi
 
+if [ -n "${NETDATA_DISABLE_CLOUD}" ]; then
```

Comment:
Let's move the checking of cloud/claiming options in a separate function to allow us to understand the high-level operations we are performing in the outer scope of the script.

Reply:
I’m honestly not sure how much this would help. Everything from this up through the end of the next statement is all one logical block, so putting it in a function would just move the code and require following the function call to find the reference.

That said, we could (and probably should) merge the two if statements here. If the `--disable-cloud` option was set the next two condition checks are pointless, so they should probably just be `elif` statements on the end of the `DISABLE_CLOUD`conditional block.